mitodl/mitxpro
# feat: added a celery task to delete expired basket

### What are the relevant tickets?
https://github.com/mitodl/mitxpro/issues/2827

### Description (What does it do?)
This PR adds a celery task which will run daily at midnight and will delete the `Baskets` which had not been modified for 15 days along with the `BasketItems`, `CourseRunSelection` and `CouponSelection`
The Baskets are also deleted on order completion, but there are scenarios where a Basket is created and the order is not completed ( see the ticket for more details ). This case shall be handled by the celery task above.
 
### Screenshots (if appropriate):


### How can this be tested?
1 In the .env file set the value `BASKET_EXPIRY_DAYS=0`
2 For any course, open the checkout page and don't complete the order. ( You can leave this step if you have existing basket and basket items )
3 Open the mitxpro shell by running `python manage.py shell` in the mitxpro directory
4 Run the following code in the shell:
```
>>> from ecommerce.models import Basket, BasketItem
>>> Basket.objects.all().count()
< number of existing baskets >
>>> BasketItem.objects.all().count()
< number of existing basket items >
```
5 Now run the following code
```
>>> from ecommerce.tasks import delete_expired_baskets
>>> delete_expired_baskets.delay()
``` 
6 In the celery logs, see the logs of this task. The task should be executed successfully and should show some useful logs.
7 Run the code in step 4 again. This time, the count will return 0

### Additional Context
